Makefile

```
@@ -13,12 +14,22 @@ PROTOC := $(LOCALBIN)/protoc
 PROTOC_GEN_GO := $(LOCALBIN)/protoc-gen-go
 PROTOC_GEN_GRPC := $(LOCALBIN)/protoc-gen-go-grpc
 PROTOC_GEN_DEEPCOPY := $(LOCALBIN)/protoc-gen-golang-deepcopy
+CONTROLLER_GEN := $(LOCALBIN)/controller-gen
 
 PROTOBUF_API_DIR := $(PROJECT_DIR)/api/proto/federation
 PROTOBUF_API_SRC := $(shell find $(PROTOBUF_API_DIR) -type f -name "*.proto")
 API_GEN_DIR=$(PROJECT_DIR)/internal/api
 PROTOBUF_GEN := $(shell find $(API_GEN_DIR) -type f -name "*.go")
 
+DEEP_COPY_DIR := $(PROJECT_DIR)/api/v1alpha1
+DEEP_COPY_SRC := $(shell find $(DEEP_COPY_DIR) -type f -not -name "*deepcopy.go")
+DEEP_COPY_GEN := $(DEEP_COPY_DIR)/zz_generated.deepcopy.go
+
+CRD_SRC_DIR := $(PROJECT_DIR)/api/v1alpha1
+CRD_SRC := $(shell find $(CRD_SRC_DIR) -type f -name "*.go")
+CRD_GEN_DIR := $(PROJECT_DIR)/chart/crds
+CRD_GEN := $(shell find $(CRD_GEN_DIR) -type f -name "*.yaml")
```

I think we can simplify entire generation to one command, first we can remove `DEEP_COPY_*`

```suggestion
CRD_SRC_DIR := $(PROJECT_DIR)/api/v1alpha1
CRD_SRC := $(shell find $(CRD_SRC_DIR) -type f -name "*.go")
CRD_GEN_DIR := $(PROJECT_DIR)/chart/crds
CRD_GEN := $(shell find $(CRD_GEN_DIR) -type f -name "*.yaml")
```